dxr/plugins/python/indexers.py

```diff
@@ -122,12 +124,84 @@ def file_to_index(self, path, contents):
                                self.derived_classes, self.names)
 
 
-class FileToIndex(FileToIndexBase):
-    needle_types = {
-        ast.ClassDef: 'py_type',
-        ast.FunctionDef: 'py_function',
-    }
+class IndexingNodeVisitor(ast.NodeVisitor):
+    """Node visitor that walks through the nodes in an abstract syntax
+    tree and finds interesting things to index.
+
+    """
+
+    def __init__(self, file_to_index):
+        self.file_to_index = file_to_index
+        self.function_call_stack = []
+        self._needles = None
```

Comment:
This doesn't strike me as any more private that the above 2. Let's be consistent, one way or the other.